### PR TITLE
WIP for not removing newlines on invert if

### PR DESCRIPTION
**Pull request by jmarolf**
fixes https://github.com/dotnet/roslyn/issues/42715

Still need to implement the VB side of this change

### CODE REVIEW

**Comment by jmarolf** [comment]
CC: @m-redding @JavierMatosD

File: src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -172,7 +174,8 @@ protected override StatementSyntax AsEmbeddedStatement(IEnumerable<StatementSynt
             IfStatementSyntax ifNode,
             SyntaxNode condition,
             StatementSyntax trueStatement,
-            StatementSyntax falseStatementOpt = null)
+            StatementSyntax falseStatementOpt = null,
+            bool shouldAddElasticTrivia = false)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs:178]: fwiw, i don't like optional args in internal apis (esp. feature=specific apis) as they are too easy to miss updating necessary callers).

File: src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -201,6 +204,10 @@ protected override StatementSyntax AsEmbeddedStatement(IEnumerable<StatementSynt
 
                 updatedIf = updatedIf.WithElse(elseClause);
             }
+            else if (!isSingleLine && shouldAddElasticTrivia)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs:207]: nit: rename to shouldAddElasticNewLine.  There is the concept of non-newline-elastictrivia, so this parameter is a bit unclear.

File: src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs [hunk 3]
```diff
@@ -255,5 +262,8 @@ static bool IsSingleStatement(StatementSyntax statement)
                 return true;
             }
         }
+
+        protected override bool ShouldAddElasicTrivia(IEnumerable<StatementSyntax> statementsAfterIf)
+            => CSharpSyntaxFacts.Instance.IsEndOfLineTrivia(statementsAfterIf.First().GetLeadingTrivia().First());
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs:267]: `first()` could throw on both of those right?

File: src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs [hunk 4]
```diff
@@ -595,6 +599,8 @@ private ImmutableArray<StatementRange> GetSubsequentStatementRanges(TIfStatement
             }
         }
 
+        protected abstract bool ShouldAddElasicTrivia(IEnumerable<TStatementSyntax> statementsAfterIf);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs:602]: place abstracts below constructor, or below fields if no constructor plz :)

**Comment by jmarolf** [comment]
@sigmachirality I am going to close this PR. Feel free to open a new one when you are ready for review

**Pull request closed without merging**